### PR TITLE
add a `touch` command

### LINKED ISSUE #1357: Add touch command
Since there are already implementations of some commands such as `mv` `cp` etc, I think it would be also great to add the `touch` command.
Even in it's most simple stage it should support:
1. Creating an empty file (or even multiple empty files)
2. Updating access and modification time of a file

Creating an empty file is a simple piece of code but changing the access-modification times are a little bit harder because Windows handles metadata differently than Linux. We can consider using the crate [filetime](https://github.com/alexcrichton/filetime) since it's cross-platform and may make things easier.

### PR DESCRIPTION
**Pull request by JesterOrNot**
Add a touch command!

Fixes #1357

### CODE REVIEW

File: src/commands/touch.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        raw_args: &RawCommandArgs,
+        _input: Value,
+    ) -> Result<OutputStream, ShellError> {
+        call_info
+            .process(&raw_args.shell_manager, raw_args.ctrl_c.clone(), touch)?
+            .run()
+    }
+}
+fn touch(args: TouchArgs, _context: &RunnablePerItemContext) -> Result<OutputStream, ShellError> {
+    match OpenOptions::new()
```
Comment (jonathandturner) [inline src/commands/touch.rs]: Can you add a test that this creates a file?

Comment (JesterOrNot) [reply src/commands/touch.rs]: How would one write a test for that?

Comment (jonathandturner) [reply src/commands/touch.rs]: You can look through the tests we have now to get a feel for how they work. Really all you'll need to do is `touch mytestfile` and then make sure that the file exists after the test runs

Comment (JesterOrNot) [reply src/commands/touch.rs]: Done

**Comment by jonathandturner** [comment]
Looks good!

**Pull request merged**